server-plugins/chunter-resources/src/index.ts

```diff
@@ -261,10 +271,51 @@ export async function ChunterTrigger (tx: Tx, control: TriggerControl): Promise<
   return res.flat()
 }
 
+/**
+ * @public
+ */
+export async function DMTrigger (tx: Tx, control: TriggerControl): Promise<Tx[]> {
+  if (tx._class !== core.class.TxCollectionCUD) return []
+  const hierarchy = control.hierarchy
+  const ctx = tx as TxCollectionCUD<ChunterSpace, Message>
+  if (!hierarchy.isDerived(ctx.tx.objectClass, chunter.class.Message)) {
+    return []
+  }
+  const actualTx = TxProcessor.extractTx(tx)
+  if (actualTx._class !== core.class.TxCreateDoc) {
+    return []
+  }
+  const doc = TxProcessor.createDoc2Doc(actualTx as TxCreateDoc<ChunterMessage>)
+  const dms = await control.findAll(chunter.class.DirectMessage, { _id: doc.space })
```

Comment:
Why???

Reply:
Doc is used to get the content message and DirectMessage to verify that we have a new message only in DMs